rllib/policy/policy.py

```diff
@@ -961,6 +966,28 @@ def set_state(self, state: PolicyState) -> None:
             state: The new state to set this policy to. Can be
                 obtained by calling `self.get_state()`.
         """
+        if "policy_spec" in state:
+            policy_spec = state["policy_spec"]
+            # Assert spaces remained the same.
+            if (
+                space_from_dict(policy_spec["observation_space"])
+                != self.observation_space
+            ):
+                logger.warning(
```

Comment:
I feel like we can error out here already and it will be easier to debug for anyone attempting to restore a "bad" policy here.

Reply:
You are right. However, I do see one possibility that maybe the observation space changed because the user is using a connector that "fixes" that new observation space?
So the policy could have originally been trained with observation space A, but then the user wanted to switch to observation space B, but still use the trained weights (and the original config/checkpoint). Something like this. That's why I only want to warn here and not error out.
The creation of the policy would still work ok, then.